portal/static/js/portal_wrapper/SessionMonitor.js

```diff
@@ -93,24 +118,42 @@ var SessionMonitorObj = function() { /* global $ */
                     },
                     onbeforetimeout: function() {},
                     ontimeout: function() {
+                        //check if session time is up before actually logging out user
+                        if (!l.isTimeExceeded()) {
+                            return;
+                        }
                         if (typeof(sessionStorage) !== "undefined") {
                             sessionStorage.clear();
                         }
+                        console.log("Time exceeded, logging out...");
+                        l.timeout();
+                    },
+                    timeout: function() {
+                        // this should communicate to any open browser tab 
                         l.setTimeoutStorage();
+                        // remove session start timestamp from localStorage
+                        l.removeTimeOnLoad();
                         window.location.href = l.timeoutUrl;
                     }
                 };
 
                 function s() {
                     $.when(l.onbeforetimeout()).always(l.ontimeout);
                 }
-
+                // this function is called each time on page load
                 function e() {
-                    var n = l.sessionLifetime - l.timeBeforeWarning;
+                    var lifeTime = l.sessionLifetime && l.sessionLifetime > 0 ? l.sessionLifetime : 1800000; // default to 30 minutes in miliseconds;
```

Comment:
the comments are very helpful, but breaking down how something was calculated can make it more obvious
```suggestion
                    var lifeTime = l.sessionLifetime && l.sessionLifetime > 0 ? l.sessionLifetime : 30*60*1000; // default to 30 minutes in miliseconds;
```